extras/create_snapshots_reports_scorecard.py

```diff
@@ -215,49 +225,147 @@ def create_weekly_snapshots(db, cyhy_db_section):
+        },
+        {"_id": 1},
+    ):
+        report_org_descendants.update(db.RequestDoc.get_all_descendants(i["_id"]))
+
+    # Create the list of snapshots to generate by removing
+    # report_org_descendants (their snapshots will be created when their
+    # parent org's snapshot is created)
+    return sorted(list(set(reports_to_generate) - report_org_descendants))
+
+
+def make_list_chunks(my_list, num_chunks):
+    """Split a list into a specified number of smaller lists."""
+    for i in range(0, num_chunks):
+        yield my_list[i::num_chunks]
+
+
+def create_snapshot(db, cyhy_db_section, org_id, use_only_existing_snapshots):
+    """Create a snapshot for a specified organization."""
+    snapshot_start_time = time.time()
+
+    snapshot_command = ["cyhy-snapshot", "--section", cyhy_db_section, "create"]
+
+    if use_only_existing_snapshots:
+        snapshot_command.extend(["--use-only-existing-snapshots", org_id])
+    else:
+        snapshot_command.append(org_id)
+
+    snapshot_process = subprocess.Popen(
+        snapshot_command,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    # Confirm the snapshot creation
+    data, err = snapshot_process.communicate("yes")
+
+    snapshot_duration = time.time() - snapshot_start_time
+    with sd_lock:
+        snapshot_durations.append((org_id, snapshot_duration))
+
+    # Determine org's descendants for logging below
+    org_descendants = list()
+    if not use_only_existing_snapshots:
+        if snapshot_process.returncode == 0:
+            org_descendants = db.SnapshotDoc.find_one(
+                {"latest": True, "owner": org_id}
+            )["descendants_included"]
+        else:
+            # Since snapshot creation failed, we must use this (slower)
+            # method of finding all descendants
+            org_descendants = db.RequestDoc.get_all_descendants(org_id)
+
+    if snapshot_process.returncode == 0:
+        logging.info("Successful snapshot: %s (%.2f s))", org_id, snapshot_duration)
+        with ss_lock:
+            successful_snapshots.append(org_id)
+            if org_descendants and not use_only_existing_snapshots:
+                logging.info(
+                    " - Includes successful descendant snapshot(s): %s", org_descendants
+                )
+                successful_snapshots.extend(org_descendants)
+    else:
+        logging.error("Unsuccessful snapshot: %s", org_id)
+        with fs_lock:
+            failed_snapshots.append(org_id)
+            if org_descendants and not use_only_existing_snapshots:
+                logging.error(
+                    " - Unsuccessful descendant snapshot(s): %s", org_descendants,
+                )
+                failed_snapshots.extend(org_descendants)
+        logging.error("Stderr failure detail: %s %s", data, err)
+    return snapshot_process.returncode
+
+
+def create_snapshots(org_list, db, cyhy_db_section):
```

Comment:
I'm not a fan of `create_snapshot()`/`create_snapshots()` because of their similarity. Would something like `create_snapshots_from_list()` be more at-a-glance readable?

Reply:
I agree - this would increase usability to disambiguate the two and keep users from having to rely on the help text to tell them apart.

Reply:
Excellent suggestion- see https://github.com/cisagov/cyhy-reports/pull/61/commits/36a4128ecbbb6f9373bc56924b964c84a316a91d.